build_runner_core/lib/src/util/constants.dart

```diff
@@ -19,7 +19,7 @@ final errorCachePath =
     p.url.join(cacheDir, _scriptHashFor(_scriptPath), 'error_cache');
 
 final String _scriptPath = Platform.script.scheme == 'file'
-    ? Platform.script.toFilePath()
+    ? p.relative(Platform.script.toFilePath(), from: p.current)
```

Comment:
This is probably an existing problem... but I think we might get difference hashes here for the same relative path between Windows and posix. Should we normalize that?

Reply:
hmmm, good point, `p.url.join(p.split(...))`?